internal/alb/lb/attributes.go

```diff
@@ -0,0 +1,204 @@
+	AccessLogsS3Enabled bool
+
+	// AccessLogsS3Bucket: access_logs.s3.bucket - The name of the S3 bucket for the access logs.
+	// This attribute is required if access logs are enabled. The bucket must
+	// exist in the same region as the load balancer and have a bucket policy
+	// that grants Elastic Load Balancing permissions to write to the bucket.
+	AccessLogsS3Bucket string
+
+	// AccessLogsS3Prefix: access_logs.s3.prefix - The prefix for the location in the S3 bucket
+	// for the access logs.
+	AccessLogsS3Prefix string
+
+	// IdleTimeoutTimeoutSeconds: idle_timeout.timeout_seconds - The idle timeout value, in seconds. The
+	// valid range is 1-4000 seconds. The default is 60 seconds.
+	IdleTimeoutTimeoutSeconds int64
+
+	// RoutingHTTP2Enabled: routing.http2.enabled - Indicates whether HTTP/2 is enabled. The value
+	// is true or false. The default is true.
+	RoutingHTTP2Enabled bool
+}
+
+func NewAttributes(attrs []*elbv2.LoadBalancerAttribute) (a *Attributes, err error) {
+	a = &Attributes{
+		DeletionProtectionEnabled: DeletionProtectionEnabled,
+		AccessLogsS3Enabled:       AccessLogsS3Enabled,
+		AccessLogsS3Bucket:        AccessLogsS3Bucket,
+		AccessLogsS3Prefix:        AccessLogsS3Prefix,
+		IdleTimeoutTimeoutSeconds: IdleTimeoutTimeoutSeconds,
+		RoutingHTTP2Enabled:       RoutingHTTP2Enabled,
+	}
+	for _, attr := range attrs {
+		switch aws.StringValue(attr.Key) {
+		case DeletionProtectionEnabledString:
+			a.DeletionProtectionEnabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		case AccessLogsS3EnabledString:
+			a.AccessLogsS3Enabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		case AccessLogsS3BucketString:
+			a.AccessLogsS3Bucket = aws.StringValue(attr.Value)
+		case AccessLogsS3PrefixString:
+			a.AccessLogsS3Prefix = aws.StringValue(attr.Value)
+		case IdleTimeoutTimeoutSecondsString:
+			a.IdleTimeoutTimeoutSeconds, err = strconv.ParseInt(aws.StringValue(attr.Value), 10, 64)
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+			if a.IdleTimeoutTimeoutSeconds < 1 || a.IdleTimeoutTimeoutSeconds > 4000 {
+				return a, fmt.Errorf("%s must be within 1-4000 seconds", aws.StringValue(attr.Key))
+			}
+		case RoutingHTTP2EnabledString:
+			a.RoutingHTTP2Enabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		default:
+			return a, fmt.Errorf("invalid load balancer attribute %s", aws.StringValue(attr.Key))
```

Comment:
i think it should be a warning via log here. Or add another params to the NewAttribute to indicate whether we are build from user's annotation or elbv2's response.

Since AWS elbv2 might adds new default attributes to lb, and our controller will error. We should be forward compatible too

Reply:
either way the error should be pushed up to fail the reconcile and it will be added with an event and error log there